classifier/UnitClassifier.js

```diff
@@ -3,17 +3,23 @@ const UnitClassification = require('../classification/UnitClassification')
 
 class UnitClassifier extends WordClassifier {
   each (span) {
-    // skip spans which do not contain numbers
-    if (!span.contains.numerals) { return }
-
-    if (/^\d+$/.test(span.body)) {
-      let prev = span.graph.findOne('prev')
-
-      // Unit must be preceded by unit type
-      if (!prev || !prev.classifications.hasOwnProperty('UnitTypeClassification')) {
-        return
-      }
+    const prev = span.graph.findOne('prev')
+    const hasPrevUnitToken = prev && prev.classifications.hasOwnProperty('UnitTypeClassification')
 
+    // all numbers or a single letter need to follow a unit type to be a unit
+    // ex: suite a, apt 22
+    if (
+      hasPrevUnitToken && (
+        // all numbers
+        /^#?\d+$/.test(span.body) ||
+        // single letter
+        /^#?[A-Za-z]$/.test(span.body) ||
+        // numbers followed by letter
+        /^#?\d+[A-Za-z]$/.test(span.body) ||
+        // letter followed by numbers
+        /^#?[A-Za-z]\d+$/.test(span.body)
```

Comment:
It may be pointless, but I did a little benchmark on regex, I think we can reduce the number of regex without making the code more complicated.

```js
const span = { body: '#a10' }
let start = new Date()
// Original
for (let i = 0; i < 100000000; i++) {
 let res = /^#?\d+$/.test(span.body) ||
        // single letter
        /^#?[A-Za-z]$/.test(span.body) ||
        // numbers followed by letter
        /^#?\d+[A-Za-z]$/.test(span.body) ||
        // letter followed by numbers
        /^#?[A-Za-z]\d+$/.test(span.body)
}
console.log(new Date() - start) // 16085ms

start = new Date()
// One regex
for (let i = 0; i < 100000000; i++) {
 let res = /^#?(\d+[a-zA-Z]?|[a-zA-Z]\d+?)$/.test(span.body)
}
console.log(new Date() - start) // 5489ms

start = new Date()
// two regex
for (let i = 0; i < 100000000; i++) {

 let res = /^#?(\d+[A-Za-z]?)$/.test(span.body) || // Starting with number with an optional letter
        /^#?([A-Za-z]\d+?)$/.test(span.body) // Starting with letter with an optional number
}
console.log(new Date() - start) // 8776ms
```

As you can see, using only one regex is 3x faster ! But this may be less readable for a newbie... IMO using two regex is a good compromise.